experimenter/experimenter/experiments/models.py

```diff
@@ -1265,11 +1265,15 @@ class VersionedSchemaRange:
         # Any versions in the requested range that do not support the schema.
         unsupported_versions: list["NimbusFeatureVersion"]
 
+        # Any versions in the requested range that support the schema.
+        supported_versions: list["NimbusFeatureVersion"]
```

Comment:
Used for fetching FML errors--we can just iterate over this list

Reply:
We can iterate over the schemas instead, as they have the versions attached, which is what you are already doing in serializers.py (i.e., this code is unused).

Reply:
I think it makes more sense to me to iterate over the unsupported versions, then to iterate over the supported versions. Something like this: 

```
        errors = []

        for version in schemas_in_range.unsupported_versions:
            errors.extend(
                NimbusConstants.ERROR_FEATURE_CONFIG_UNSUPPORTED_IN_VERSION.format(
                    feature_config=feature_config,
                    version=version,
                )
            )

        for version in schemas_in_range.supported_versions:
            if fml_errors := loader.get_fml_errors(blob, feature_config.slug, version):
                errors.extend(
                    [
                        f"{NimbusExperiment.ERROR_FML_VALIDATION}: {e.message} at line "
                        f"{e.line+1} column {e.col} at version {version}"
                        for e in fml_errors
                    ]
                )

        return errors
```

What do you think?

Reply:
`schemas_in_range.supported_versions` is equivalent to `[s.version for s in shemas_in_range.schemas`.

What I'm suggesting is just changing your second for loop to:

```python
        for schema in schemas_in_range.schemas: # <--
            version = schema.version            # <--
            if fml_errors := loader.get_fml_errors(blob, feature_config.slug, version):
                errors.extend(
                    [
                        f"{NimbusExperiment.ERROR_FML_VALIDATION}: {e.message} at line "
                        f"{e.line+1} column {e.col} at version {version}"
                        for e in fml_errors
                    ]
                )
````

Reply:
If you absolutely *need* supported_versions, it should be a computed property:

```python
@property
def supported_versions(self) -> typing.Generator[NimbusFeatureVersion, None, None]:
    for schema in self.schemas:
        yield schema.version
```

Reply:
Ahh I see what you mean. The `schemas` list only contains supported versions, and filters out the `unsupported` into the separate list. Instead of having `supported`, I'm going to update the description of `schemas` to make this more explicit and remove the `supported` list 👍